MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/IconSizeChangeOnClickActivity.java

```diff
@@ -0,0 +1,218 @@
+    Style style = mapboxMap.getStyle();
+    if (style != null) {
+      final SymbolLayer selectedMarkerSymbolLayer =
+        (SymbolLayer) style.getLayer("selected-marker-layer");
+
+      final PointF pixel = mapboxMap.getProjection().toScreenLocation(point);
+      List<Feature> features = mapboxMap.queryRenderedFeatures(pixel, "marker-layer");
+      List<Feature> selectedFeature = mapboxMap.queryRenderedFeatures(
+        pixel, "selected-marker-layer");
+
+      if (selectedFeature.size() > 0 && markerSelected) {
+        return false;
+      }
+
+      if (features.isEmpty()) {
+        if (markerSelected) {
+          deselectMarker(selectedMarkerSymbolLayer);
+        }
+        return false;
+      }
+
+      GeoJsonSource source = style.getSourceAs("selected-marker");
+      if (source != null) {
+        source.setGeoJson(FeatureCollection.fromFeatures(
+          new Feature[]{Feature.fromGeometry(features.get(0).geometry())}));
+      }
+
+      if (markerSelected) {
+        deselectMarker(selectedMarkerSymbolLayer);
+      }
+      if (features.size() > 0) {
+        selectMarker(selectedMarkerSymbolLayer);
+      }
+    }
+    return true;
+  }
+
+  private void selectMarker(final SymbolLayer iconLayer) {
+    ValueAnimator markerAnimator = new ValueAnimator();
+    markerAnimator.setObjectValues(1f, 2f);
+    markerAnimator.setDuration(300);
+    markerAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+
+      @Override
+      public void onAnimationUpdate(ValueAnimator animator) {
+        iconLayer.setProperties(
+          PropertyFactory.iconSize((float) animator.getAnimatedValue())
+        );
+      }
+    });
+    markerAnimator.start();
+    markerSelected = true;
+  }
+
+  private void deselectMarker(final SymbolLayer iconLayer) {
+    ValueAnimator markerAnimator = new ValueAnimator();
+    markerAnimator.setObjectValues(2f, 1f);
+    markerAnimator.setDuration(300);
+    markerAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+
+      @Override
+      public void onAnimationUpdate(ValueAnimator animator) {
+        iconLayer.setProperties(
+          PropertyFactory.iconSize((float) animator.getAnimatedValue())
+        );
+      }
+    });
+    markerAnimator.start();
+    markerSelected = false;
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    if (mapboxMap != null) {
+      mapboxMap.removeOnMapClickListener(this);
+    }
+    mapView.onDestroy();
```

Comment:
we aren't stopping animations in this example so we can potentially crash with selecting a symbol and closing the activity quickly after.

Reply:
Ok, https://github.com/mapbox/mapbox-android-demo/pull/1030/commits/4d4e13a314aca56105a528ec50037e1be0999afb adds animation cancellation in `onDestroy()`.

Reply:
I closed the example during the icon size animation. No issues.